wcomponents-theme/src/main/js/wc/file/accepted.mjs

```diff
@@ -23,45 +25,64 @@ import getMimeType from "wc/file/getMimeType.mjs";
  */
-export default function accepted(element) {
+export default function accepted(element, fileInfo) {
 	let acceptedType = element.accept;
-	if (acceptedType) {
-		let acceptedTypes = acceptedType.split(",");
-		return getMimeType(element).every((/** @type {module:wc/file/getMimeType~fileType} */fileType) => {
-			let mimeType = fileType.mime || "",
-				extension = fileType.ext ? (`.${fileType.ext}`) : "",
-				passed = (!mimeType && !extension) ||
-						(extension && acceptedType.indexOf(extension) >= 0) ||
-						(mimeType && acceptedType.indexOf(mimeType) >= 0);
-			if (!passed) {  // maybe there is a case difference OR it's a wildcard mimetype or there is some whitespace to be trimmed?
-				for (let i = 0; i < acceptedTypes.length; i++) {
-					let next = acceptedTypes[i].toLowerCase();
-					next = next.trim();
-					if (extension) {
-						extension = extension.toLowerCase();
-						if (extension === next) {
-							passed = true;
-							break;
-						}
-					}
-					if (mimeType) {
-						mimeType = mimeType.toLowerCase();
-						if (mimeType === next) {
-							passed = true;
-							break;
-						} else if (next.indexOf("*") === next.length - 1) {
-							next = next.substring(0, next.length - 1);
-							if (mimeType.indexOf(next) === 0) {
-								passed = true;
-								break;
-							}
-						}
-					}
+
+	if (!acceptedType) return true;
+
+	let acceptedTypes = acceptedType.split(",");
+
+	const checkFileBasic = function (mimeType, extension) {
+		return (!mimeType && !extension) ||
+			(extension && acceptedType.indexOf(extension) >= 0) ||
+			(mimeType && acceptedType.indexOf(mimeType) >= 0);
+	}
+
+	const compareMime = function (actual, reference) {
+		if (actual === reference) {
+			return true;
+		} else if (reference.indexOf("*") === reference.length - 1) {
+			reference = reference.substring(0, reference.length - 1);
+			if (actual.indexOf(reference) === 0) {
+				return true;
+			}
+		}
+		return false;
+	};
+
+	const checkFileArray = function (/** @type {module:wc/file/getMimeType~fileType} */fileType) {
+		let mimeType = fileType.mime || "",
+			extension = fileType.ext ? (`.${fileType.ext}`) : "",
+			passed = checkFileBasic(mimeType, extension);
+
+		if (passed) {
+			return true; // return straight away after passing first check
+		}
+
+		// maybe there is a case difference OR it's a wildcard mimetype or there is some whitespace to be trimmed?
+		for (let i = 0; i < acceptedTypes.length; i++) {
+			let next = acceptedTypes[i].toLowerCase();
+			next = next.trim();
+			if (extension) {
+				extension = extension.toLowerCase();
+				if (extension === next) {
+					passed = true;
+					break;
 				}
 			}
-			return passed;
-		});
-	}
-	return true;
+			if (mimeType) {
+				mimeType = mimeType.toLowerCase();
+				passed = compareMime(mimeType, next);
+				if (passed) {
+					break;
+				}
+			}
+		}
+		return passed;
+	};
+
+	if (fileInfo) return getMimeType(fileInfo).every(checkFileArray);
```

Comment:
While this works fine and there is nothing at all wrong with it, we inherited a style format that insists on curly braces even for single statements. The thinking behind such a rule is that it prevents future bugs, as someone comes along and adds another statement, forgetting to add the curly braces.

I'm not saying I agree with it, but that's how the rest of the codebase is written, so I suggest you add the curly braces to this one too.